contracts/stellar-interchain-token-service/src/abi.rs

```diff
@@ -776,25 +798,21 @@ mod tests {
             result,
             Err(ContractError::InvalidTokenManagerType)
         ));
+    }
 
-        // Test decoding fails with NativeInterchainToken
-        // First create a valid LinkToken payload with LockUnlock type
-        let valid_link_token = types::Message::LinkToken(types::LinkToken {
-            token_id: BytesN::from_array(&env, &[0u8; 32]),
-            token_manager_type: types::TokenManagerType::LockUnlock,
-            source_token_address: Bytes::from_hex(&env, "00"),
-            destination_token_address: Bytes::from_hex(&env, "00"),
-            params: None,
-        });
-
-        let mut payload = valid_link_token.abi_encode(&env).unwrap().to_alloc_vec();
-
-        // Manually modify the tokenManagerType field to 0 (NativeInterchainToken)
-        // The structure is: messageType(32) + tokenId(32) + tokenManagerType(32) + ...
-        // So tokenManagerType starts at byte 64
-        for i in 64..96 {
-            payload[i] = 0;
-        }
+    #[test]
+    fn link_token_decode_fails_with_native_interchain_token_type() {
+        let link_token = LinkToken {
+            messageType: MessageType::LinkToken.into(),
+            tokenId: FixedBytes::<32>::new([0u8; 32]),
+            tokenManagerType: U256::from(0u32), // NativeInterchainToken
```

Comment:
use `TokenManagerType::NativeInterchainToken as u32` explicitly instead of hardcoding